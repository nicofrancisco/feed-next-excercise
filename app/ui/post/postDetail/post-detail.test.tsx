import '@testing-library/jest-dom';
import React from "react";
import { render, screen } from "@testing-library/react";
import { expect } from 'vitest';
import PostDetail from './post-detail'
import { Post, User } from '../../../lib/types'

const mockPost: Post = {
  id: 1,
  userId:1,
  title: 'Test Post Title',
  body: 'This is the body of the test post.'
}

const mockUser: User = {
  id: 1,
  name: 'Test User',
  username: 'testuser',
  email: 'testuser@example.com',
  address: {
    street: '123 Test St',
    suite: 'Apt 1',
    city: 'Test City',
    zipcode: '12345',
    geo: {
      lat: '0.0000',
      lng: '0.0000'
    }
  },
  company: {
    name: 'Test Company',
    catchPhrase: 'Test Catchphrase',
    bs: 'Test BS'
  },
  phone: '123-456-7890',
  website: 'www.testuser.com'
}

describe('PostDetail', () => {
    it('renders correctly', () => {
        const { asFragment } = render(<PostDetail post={mockPost} user={mockUser} />)
        expect(asFragment()).toMatchSnapshot()
    })

    it('renders the user name', () => {
        render(<PostDetail post={mockPost} user={mockUser} />)
        expect(screen.getByText('Test User')).toBeInTheDocument()
    })
    
    it('renders the post title', () => {
        render(<PostDetail post={mockPost} user={mockUser} />)
        expect(screen.getByText('Test Post Title')).toBeInTheDocument()
    })
    
    it('renders the user image', () => {
        render(<PostDetail post={mockPost} user={mockUser} />)
        const userImage = screen.getByAltText('user-avatar')
        expect(userImage).toBeInTheDocument()
        expect(userImage).toHaveAttribute('src')
    })
})