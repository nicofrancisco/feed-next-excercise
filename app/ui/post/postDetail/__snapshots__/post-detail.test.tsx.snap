// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PostDetail > renders correctly 1`] = `
<DocumentFragment>
  <div
    class="relative flex flex-col gap-4 animate-fadeInUp"
  >
    <div
      class="flex flex-col items-center gap-4"
    >
      <a
        href="/feed"
      >
        <svg
          class="stroke-gray-500 absolute mt-4 top-0 left-0 size-6 hover:stroke-gray-900"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15.75 19.5 8.25 12l7.5-7.5"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </a>
      <img
        alt="user-avatar"
        class="w-24 h-24 rounded-full bg-gray-50"
        loading="lazy"
        src="https://api.dicebear.com/8.x/adventurer/svg?seed=Test User"
      />
      <span
        class="text-xl font-bold leading-6 text-gray-800"
      >
        Test User
      </span>
    </div>
    <article
      class="flex flex-col gap-4"
    >
      <h2
        class="text-lg font-semibold leading-6 text-gray-700 capitalize"
      >
        Test Post Title
      </h2>
      <p
        class="text-md leading-6 text-gray-900"
      >
        This is the body of the test post.
        <br />
        <br />
        This is the body of the test post.. This is the body of the test post..
      </p>
    </article>
  </div>
</DocumentFragment>
`;
